src/Plugin/resource/DataProvider/DataProviderInterface.php

```diff
@@ -192,4 +192,12 @@ public function setResourcePath($resource_path);
    */
   public function getResourcePath();
 
+  /**
+   * Returns the metadata collection.
```

Comment:
I'd add a longer explanation like in in the OP of the PR, to explain how/ when it can be used.